ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/events/TransactionIndexingSpec.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events
+
+import java.time.Instant
+
+import com.daml.ledger
+import com.daml.ledger.participant.state.v1.Offset
+import com.daml.lf.transaction.BlindingInfo
+import com.daml.lf.transaction.test.TransactionBuilder
+import org.scalatest.{Matchers, WordSpec}
+
+final class TransactionIndexingSpec extends WordSpec with Matchers {
+  "TransactionIndexing" should {
```

Comment:
It would probably make sense to add a test case with a fetch, to make sure we're not missing those.

Reply:
The idea of this change is to remove `Fetch` and `LookupByKey` nodes and have everything rely instead only `blindingInfo`; this PR doesn't fix an existing problem but completes the removal altogether. Since the goal is such a removal, I'm not sure about what to test regarding fetches. Can you expand, e.g. what would you like to check in addition?

Reply:
If you are removing all fetches in all cases and not only optionally, then yes.

Reply:
Yes, all of them, always (on the committer, after validating/executing and saving blinding info in a new field).

Reply:
Does this mean that the committer now effectively _must_ send the blinding? I'm not sure you can get the correct blinding with a partial view of the transaction.

Reply:
> Does this mean that the committer now effectively must send the blinding?

Yes, with this this PR it has to and it has done so since #8009.

> I'm not sure you can get the correct blinding with a partial view of the transaction.

I'm not sure I understand, which partial view do you mean and which blinding would be missing?

Reply:
> I'm not sure I understand, which partial view do you mean and which blinding would be missing?

If you don't send fetch nodes, can you still run blinding on the transaction and get back an accurate result?

> Yes, with this this PR it has to and it has done so since #8009.

Are we planning to reflect this into the types as well? Right now the function is taking an `Option[BlindingInfo]` but that should probably become a `BlindingInfo`, right?

Reply:
> If you don't send fetch nodes, can you still run blinding on the transaction and get back an accurate result?

Yes, because the blinding info is computed from those nodes before they are removed, then saved and restored on the read path.

> Are we planning to reflect this into the types as well? Right now the function is taking an Option[BlindingInfo] but that should probably become a BlindingInfo, right?

Which function do you mean? A requirement is that the old format (that doesn't have committer-pre-computed `BlindingInfo`) should still work, that's why `TransactionWriter.prepare` receives it as an `Option` and, if not present (which means old format, i.e. the `Fetch` and `LookupByKey` nodes are still there), computes it. But `TransactionIndexing.from` has to receive a `BlindingInfo` and it was never optional there.

Reply:
> A requirement is that the old format (that doesn't have committer-pre-computed BlindingInfo) should still work

Ha-ha, of course, backwards compatibility, thanks.

> Yes, because the blinding info is computed from those nodes before they are removed, then saved and restored on the read path, that gives precedence to that pre-computed blinding info (new format) over computing it from the nodes (old format) in #8012.

Allow me to re-phrase: the assumption is that one should send either

* the full transaction with no blinding (old), _or_
* the partial transaction with blinding (new).

Something like "the partial transaction without blinding" is not possible, right?

Reply:
And, in that case, blinding info has precedence.